### PR TITLE
[EncoderDecoderModel] Add support for gradient checkpointing

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

This PR adds gradient checkpointing support for EncoderDecoderModel in PyTorch.

As requested on the forum: https://discuss.huggingface.co/t/feature-request-gradient-checkpointing-for-encoderdecodermodel/25278

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for adding this!
